modules/ROOT/pages/planning-and-tuning/query-tuning.adoc

```diff
@@ -303,3 +303,29 @@ CYPHER replan=force EXPLAIN MATCH ...
 
 During times of known high load, `replan=skip` can be useful to not introduce unwanted latency spikes.
 
+[role=label--new-5.21]
+[[cypher-infer-schema-parts]]
+=== Cypher infer schema parts
+
+For some queries, the planner can infer predicates such as labels or types from the graph structure, thereby enhancing its ability to estimate the number of rows each operator will produce.
+(See xref:planning-and-tuning/execution-plans.adoc#runtimes-reading-execution-plans[Understanding execution plans - Reading execution plans] for more information about the role of operators and estimated row counts in query execution plans.)
+The option `inferSchemaParts` controls the extent to which the planner should infer predicates.
+
+[options="header",cols="2m,3a"]
+|===
+| Option | Description
+
+| inferSchemaParts=off
+| No predicates are inferred.
+
+| inferSchemaParts=most_selective_label
+| Relationship types are used to infer labels on the end nodes of relationships.
```

Comment:
I would consider not writing "end nodes", as that could be confused with just the nodes the relationships are pointing _to_.

"boundary" nodes? "connected" nodes?